awx/main/utils/receptor.py

```diff
@@ -71,7 +71,12 @@ def worker_info(node_name):
         if not isinstance(remote_data, dict):
             error_list.append(f'Remote node {node_name} --worker-info output is not a YAML dict, output:{stdout}')
         else:
-            error_list.extend(remote_data.pop('Errors'))  # merge both error lists
+            error_list.extend(remote_data.pop('errors', []))  # merge both error lists
             data.update(remote_data)
 
+    # see tasks.py usage of keys
+    missing_keys = set(('runner_version', 'mem_in_bytes', 'cpu_count')) - set(data.keys())
+    if missing_keys:
+        data['errors'].append('Worker failed to return keys {}'.format(' '.join(missing_keys)))
```

Comment:
This was an error case we were missing before - if the remote executor gives no relevant keys. We mark it lost if there are errors, but there was no "errors" key returned, so it was saved as a working instance with a memory capacity of 1. That's not good, but is resolved here to my satisfaction.